glean-core/ios/Glean/Net/HttpPingUploader.swift

```diff
@@ -125,3 +149,56 @@ public class HttpPingUploader {
         }
     }
 }
+
+/// An object that can be assigned as a session delegate and will hold the callback with which to respond to
+/// glean-core with the network response
+private class SessionResponseDelegate: NSObject, URLSessionTaskDelegate {
```

Comment:
Yeah, I wasn't sure about calling `finish` from the delegate. Apple docs are "fun" sometimes. I'll tinker with this a bit, it might be fine to just call `finishTasksAndInvalidate` after queueing the task, if that allows the task to complete.

Reply:
Looks like the documentation says it does allow tasks to complete, and a little testing shows that it seems to work fine after adding the call to it after `resume()` is called on the upload task.